### DISCUSSION TITLE
How to retrieve the selected option onItemAdd

### DISCUSSION THREAD

**Discussion by trullock** [orchidjs/tom-select#143] [Q&A] [answered]
`onItemAdd`'s arguemnts are the string value and the $div element.

When ive populated my select like this:

```
options: [
		{id: 1, title: 'Spectrometer', url: 'http://en.wikipedia.org/wiki/Spectrometers'},
		{id: 2, title: 'Star Chart', url: 'http://en.wikipedia.org/wiki/Star_chart'},
		{id: 3, title: 'Electrical Tape', url: 'http://en.wikipedia.org/wiki/Electrical_tape'}
	],
```

it would be great to be able to get at the option object in the onItemAdd method, e.g. `{id: 1, title: 'Spectrometer', url: 'http://en.wikipedia.org/wiki/Spectrometers'}` is this possible, or do i have to re-find the option by value from the options list?

**Comment by oyejorge** [comment, marked as answer]
Yeah, getting the data requires:

```js
onItemAdd:function(id,$div){
   const data = this.options[id];
}
```

**Reply by trullock** [reply]
ah thats convenient, ok ta
